shell/components/form/PodAffinity.vue

```diff
@@ -36,10 +37,20 @@ export default {
       default: () => []
     },
 
-    hasNodesAndNs: {
+    overrideStore: {
+      type:    String,
+      default: ''
+    },
+
+    hasNodes: {
```

Comment:
This doesn't need to be a prop: compute from `this.nodes.length`

Changing this to a computed property will change the UX in workload pod scheduling, but I think the behavior in harvester machine-config is preferable there, too. eg if a user making a workload can't see nodes, they could still type in a topology key.